WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -195,9 +195,8 @@ final class MainTabBarController: UITabBarController {
     ///   - animated: Whether the tab switch is animated.
     ///   - completion: Invoked when switching to the tab's root screen is complete.
     func navigateToTabWithNavigationController(_ tab: WooTab, animated: Bool = false, completion: ((UINavigationController) -> Void)? = nil) {
-        if let presentedController = Self.childViewController()?.presentedViewController {
-            presentedController.dismiss(animated: true)
-        }
+        view.window?.rootViewController?.dismiss(animated: true, completion: nil)
```

Comment:
> I tested both approaches in coupons and I couldn't see any difference with this change, however I'm thinking it this could be a concern for other flows that use this same method and are using multiple view controllers in the stack (specially if modally) since we're losing controller over dismissing only the presented from the child one. Thoughts? 🤔

For me when testing this on this case, where we have two modals presented, the old approach only dismissed one (the coupon selector), leaving us with the order details presented. This new approach dismisses everything and displays the root of the navigated tab.

I cannot imagine a case where we want to navigate to a tab and leave one modal presented as it defeats the purpose of the navigation itself. I can make this cleaner though, and replace that line with `dismiss(animated: true, completion: nil)`, which has the same effect. Done in 5b7345a9461f4f3eab55c7e89aa7244f262d5586